src/MatroskaParser.c

```diff
@@ -40,20 +40,25 @@
 #include <string.h>
 #include <setjmp.h>
 
-#ifdef _WIN32
+#ifdef HAVE_ALLOCA_H
+#include <alloca.h>
+#elif defined(HAVE_MALLOC_H)
+#include <malloc.h>
```

Comment:
I can't say I know of every edge case here, but could this be accomplished by just using c++17's `__has_include` instead of extra build system logic? Even if `MatroskaParser.c` is a C file, one could define a helper function in a different C++ file. Though that could also end up being very hacky.